yfinance/utils.py

```diff
@@ -282,35 +306,48 @@ def camel2title(strings: List[str], sep: str = ' ', acronyms: Optional[List[str]
 
     # Apply str.title() to non-acronym words
     strings = [s.split(sep) for s in strings]
-    strings = [[j.title() if not j in acronyms else j for j in s] for s in strings]
+    strings = [[j.title() if not j in acronyms else j for j in s]
+               for s in strings]
     strings = [sep.join(s) for s in strings]
 
     return strings
 
 
-def _parse_user_dt(dt, exchange_tz):
-    if isinstance(dt, int):
-        # Should already be epoch, test with conversion:
-        _datetime.datetime.fromtimestamp(dt)
-    else:
-        # Convert str/date -> datetime, set tzinfo=exchange, get timestamp:
-        if isinstance(dt, str):
+def _parse_user_dt(dt, exchange_tz: Union[str, None] = None) -> Union[int, None]:
+    try:
+        if isinstance(dt, int):
+            # Should already be epoch, test with conversion:
+            dt = _datetime.datetime.fromtimestamp(dt)
+            # Convert str/date -> datetime, set tzinfo=exchange, get timestamp:
+        elif isinstance(dt, str):
             dt = _datetime.datetime.strptime(str(dt), '%Y-%m-%d')
-        if isinstance(dt, _datetime.date) and not isinstance(dt, _datetime.datetime):
+        elif isinstance(dt, _datetime.date) and not isinstance(dt, _datetime.datetime):
             dt = _datetime.datetime.combine(dt, _datetime.time(0))
-        if isinstance(dt, _datetime.datetime) and dt.tzinfo is None:
+        elif isinstance(dt, _datetime.datetime) and dt.tzinfo is None:
```

Comment:
Are you absolutely **sure** the logic hasn't changed by switching `if` -> `elif`? I deliberately used `if` for each to behave like a data pipeline.

Reply:
Yes, the behavior did not change at all. It was just a matter of semantics really. All the branches are checking the instance of dt and it seems to be more appropriate to have them at the same lever. But then again the behavior did not change

Reply:
I tested, it does change. If user provides a string the timezone isn't set. Please be careful.

Reply:
I guess we could just default it to "UTC" or whatever you think is best. If we don't set a default then we are forced to pass a timezone exchange even though the expression does not use it

for example

```python3
if isinstance(dt, int):
        # Should already be epoch, test with conversion:
        _datetime.datetime.fromtimestamp(dt)
```

what do you think should be the default? also, if you authored the function, could you write some tests so that we can track breaking changes? if you didn't author the function, I would gladly write the test cases

Reply:
Short answer: no default.

Long answer:
Presumably the `dt` value you're passing into that function is from the user, so you need to provide a timezone anyway because it's usually ISO 8601 ("2023-01-01").

Assuming UTC was cause of user experience issues before my fix - missing a day or receiving a day outside of requested range. Assuming the exchange timezone fixed this, and just makes sense.

I've started branch `clarify-datetime-handling` for answering these questions.